### DISCUSSION TITLE
Can i use ioredis on context as a util ?

### DISCUSSION THREAD

**Discussion by nandocoeg2** [vercel/next.js#59323] [Help]
### Summary

i have tried use ioredis on pages as util and it's work

![image](https://github.com/vercel/next.js/assets/97197966/4891b3bc-078f-4647-8c9c-a14bc37a0303)

but i import redis util on context it's not working. i got error like this

![image](https://github.com/vercel/next.js/assets/97197966/e2f34524-f2eb-45ca-851b-bd8044380e52)
![image](https://github.com/vercel/next.js/assets/97197966/f206c7e1-dba6-4182-9aa4-813b1e63edb2)
![image](https://github.com/vercel/next.js/assets/97197966/d1ccc529-e425-4203-81dd-2e8cb285d2d2)

**Comment by icyJoseph** [comment]
The problem is that, `PnAuthContext` will be bundled for usable in a browser. And the browser doesn't have a `dns` module, that only exists on Node.js

**Reply by nandocoeg2** [reply]
did you have an idea to pass data from redis to context provider ?

**Reply by icyJoseph** [reply]
Yeah so, getServerSideProps has this quality, that helps Next.js eliminate code used within from the browser bundle, that's why you can use it page level, and it causes no issues. 

- https://next-code-elimination.vercel.app/

What you'd need to do is, get that data in a data-fetching function, such as getServerSideProps, and send that to the client as props, and use that to feed the context provider too.

**Reply by nandocoeg2** [reply]
i already do that on pages. the question is, is it possible use getServerSideProps on PnAuthContext and send as props to context provider?
